exporter/datadogexporter/factory.go

```diff
@@ -269,11 +269,19 @@ func (f *factory) createMetricsExporter(
 	statsIn := make(chan []byte, 1000)
 	statsv := set.BuildInfo.Command + set.BuildInfo.Version
 	f.consumeStatsPayload(ctx, &wg, statsIn, statsWriter, statsv, acfg.AgentVersion, set.Logger)
-	pcfg := newMetadataConfigfromConfig(cfg)
-	metadataReporter, err := f.Reporter(set, pcfg)
-	if err != nil {
-		cancel()
-		return nil, fmt.Errorf("failed to build host metadata reporter: %w", err)
+
+	// These variables are referenced below, but only used if HostMetadata.Enabled
```

Comment:
I find this comment a bit confusing, why not define them inside the `if` block then?

Reply:
Because they're referenced below. Unless we structure the whole function, we need these variables to be defined for the rest of the function.

Reply:
I guess I find the usage of "referenced" and "used" here confusing on the comment then: from your message I take that they are "used" (mentioned in code) both if HostMetadata.Enabled and if not, right?

Reply:
Right, I guess we have different definitions of "used".

The two variables are _mentioned_ in the code below, which requires their scope to be larger than the `if` statement if we want the code to compile. This is independent of what path the code takes.

When `!HostMetadata.Enabled`, `errorMetadata` will still be _read_, to be passed to the `new(Logs|Traces|Metrics)Exporter` function. It will be `nil` in that case. But the pointer won't be _used_ later down the line, ie. no methods will be called on it.

I can reword the comment to make this clearer if you'd like.

Reply:
Ah, right, thanks! I will let you decide whether to reword it. I did find it confusing, and I don't feel like this is standard wording for this, but maybe it's just me :smile:

Reply:
I simplified the logic and the comments somewhat, tell me if you think what's going on is clearer or not